src/lando/utils/github.py

```diff
@@ -0,0 +1,83 @@
+import asyncio
+import logging
+
+import requests
+from django.conf import settings
+from simple_github import AppAuth, AppInstallationAuth
+
+from lando.main.models.repo import Repo
+
+logger = logging.getLogger(__name__)
+
+
+class GitHubAPI:
+    GITHUB_BASE_URL = "https://api.github.com"
+
+    def __init__(self, repo: Repo):
```

Comment:
Using `Repo` here is introducing a strong coupling that we can probably avoid.

We could either extract the owner and name in the runner, and pass those two arguments to the constructor, or maybe just allow the `__init__` here to just get an unparsed `Repo.url`, and let it extract the components from the URL.

This way we'd have all the GitHub logic nicely tucked in one place.